hydro_deploy/core/src/util.rs

```diff
@@ -26,60 +26,73 @@ pub async fn async_retry<T, F: Future<Output = Result<T>>>(
 }
 
 type PriorityBroadcacst = (
-    Arc<RwLock<Option<tokio::sync::oneshot::Sender<String>>>>,
-    Arc<RwLock<Vec<Sender<String>>>>,
+    Arc<Mutex<Option<tokio::sync::oneshot::Sender<String>>>>,
+    Arc<Mutex<Vec<Sender<String>>>>,
 );
 
 pub fn prioritized_broadcast<T: Stream<Item = io::Result<String>> + Send + Unpin + 'static>(
     mut lines: T,
     default: impl Fn(String) + Send + 'static,
 ) -> PriorityBroadcacst {
-    let priority_receivers = Arc::new(RwLock::new(None::<tokio::sync::oneshot::Sender<String>>));
-    let receivers = Arc::new(RwLock::new(Vec::<Sender<String>>::new()));
+    let priority_receivers = Arc::new(Mutex::new(None::<tokio::sync::oneshot::Sender<String>>));
+    let receivers = Arc::new(Mutex::new(Vec::<Sender<String>>::new()));
 
     let weak_priority_receivers = Arc::downgrade(&priority_receivers);
     let weak_receivers = Arc::downgrade(&receivers);
 
     tokio::spawn(async move {
-        'line_loop: while let Some(Result::Ok(line)) = lines.next().await {
+        while let Some(Result::Ok(line)) = lines.next().await {
             if let Some(cli_receivers) = weak_priority_receivers.upgrade() {
-                let mut cli_receivers = cli_receivers.write().await;
+                let mut cli_receivers = cli_receivers.lock().unwrap();
 
                 let successful_send = if let Some(r) = cli_receivers.take() {
                     r.send(line.clone()).is_ok()
                 } else {
                     false
                 };
+                drop(cli_receivers);
```

Comment:
(this doesn't really do anything, the mutexguard is dropped here either way, I think)